Update pong book chapters (#1679)
amethyst/amethyst: book/src/pong-tutorial/pong-tutorial-01.md

# Setting up the project

In this chapter we will go through the basics of setting up the amethyst project,

> starting logger, openning a window and preparing basic rendering setup.

Reviewer: ```suggestion
starting the logger, opening a window and preparing a simple rendering setup.
```

features = ["vulkan"]
```

> Alternatively, if you are developing on macOS, you might want to use `metal` rendering backend instead of `vulkan`. In this case you should change `features` entry in amethyst dependency.

Reviewer: ```suggestion
Alternatively, if you are developing on macOS, you might want to use the `metal` rendering backend instead of `vulkan`. In this case you should change the `features` entry in the `amethyst` dependency table.
```

(
  title: "Pong!",
  dimensions: Some((500, 500)),
)
```

> **Note:** If you have never run into Rusty Object Notation before (or RON for short), 
> it is a data storage format that mirrors Rust's syntax. Here, the
> data represents the [`DisplayConfig`][displayconf] struct. If you want to
> learn more about the RON syntax, you can visit the [official repository][ron].

This will set the default window dimensions to 500 x 500, and make the title bar
say "Pong!" instead of the sad, lowercase default of "pong".

In `main()` in `main.rs`, we will prepare the path to a file containing
the display configuration:

```rust,edition2018,no_run,noplaypen
# extern crate amethyst;
#
# use amethyst::{
#     utils::application_root_dir,
#     Error,
# };
#
# fn main() -> Result<(), Error>{
    let app_root = application_root_dir()?;
    let display_config_path = app_root.join("resources").join("display_config.ron");
#     Ok(())
# }
```

## Opening a window

> After preparing the display config, now it's time to actually use it. To do that,

Reviewer: ```suggestion
After preparing the display config, it's time to actually use it. To do that,
```

    let assets_dir = app_root.join("assets");
    let mut game = Application::new(assets_dir, Pong, game_data)?;
    game.run();
```

Here we're creating a new `WindowBundle` that uses the config we prepared above.
That bundle is being used as a part of `GameDataBuilder`, a central repository
of all the game logic that runs periodically during the game runtime.

> **Note:** We will cover systems and bundles in more details later, for now think of the
> bundle as a group of functionality that together provides certain feature to the engine.
> You will surely be writing your own bundles for your own game's features soon.

That builder is then combined with the game state struct (`Pong`), creating the overarching
Amethyst's root object: [Application][ap]. It binds the OS event loop, state machines,
timers and other core components in a central place.

Then we call `.run()` on `game` which begins the gameloop. The game will
continue to run until our `SimpleState` returns `Trans::Quit`, or when all states
have been popped off the state machine's stack.

Try compiling the code now. You should be able to see the window already.

> The contents of that window right now is undefined and up to the operating system.

Reviewer: ```suggestion
The content of that window right now is undefined and up to the operating system.
```
